src/rard/research/views/fragment.py

```diff
@@ -1010,3 +1010,136 @@ def fetch_fragments(request):
+
+        # Recreate References for new OT
+        for reference in original_original_texts[i].references.all():
+            Reference.objects.create(
+                editor=reference.editor,
+                reference_position=reference.reference_position,
+                original_text=new_original_text,
+            )
+
+        # Copy relationships to Concordances, Ap Crit & Translations
+        model_details = {
+            Concordance: {
+                "filter_name": "original_text",
+                "filter_value": original_original_texts[i],
+                "ot_fieldname": "original_text",
+            },
+            ApparatusCriticusItem: {
+                "filter_name": "object_id",
+                "filter_value": original_original_texts[i].pk,
+                "ot_fieldname": "parent",
+            },
+            Translation: {
+                "filter_name": "original_text",
+                "filter_value": original_original_texts[i],
+                "ot_fieldname": "original_text",
+            },
+        }
+
+        for model_class in model_details.keys():
+            model_info = model_details[model_class]
+            filter_name = model_info["filter_name"]
+            filter_value = model_info["filter_value"]
+            ot_fieldname = model_info["ot_fieldname"]
+
+            new_model_data = {}
+            for original_item in model_class.objects.filter(
+                **{filter_name: filter_value}
+            ):
+                for field in original_item._meta.fields:
+                    # Exclude some fields
+                    if field.name in [
+                        "id",
+                        "created",
+                        "modified",
+                        ot_fieldname,
+                        "content_type",
+                    ]:
+                        continue
+
+                    field_value = getattr(original_item, field.name)
+                    new_model_data[field.name] = field_value
+
+                new_model_data[
+                    ot_fieldname
+                ] = new_original_text  # make sure to assign new OT in place of the old one (field skipped above)
+
+                new_model = model_class.objects.create(**new_model_data)
+                # specific handling of content_type since we don't want to create a new instance
+                if hasattr(model_class, "content_type") and model_class.content_type:
+                    new_model.content_type = ContentType.objects.get(
+                        pk=original_item.content_type.pk
+                    )
+
+    # Create a new fragment with the same values as original
+    new_fragment_data = {}
+    for field in original_fragment._meta.fields:
+        # Exclude unique fields
+        if field.name in [
+            "id",
+            "created",
+            "modified",
+            "commentary",
+            "order",  # anon frags have this
+            "model",
```

Comment:
it would distinguish between af and frag but I'll try without it and see if it throws the error it did before

Reply:
<img width="559" alt="image" src="https://github.com/UCL/frrant/assets/72266023/29de6c4f-3d89-423a-91ff-1eb3507f402b">
It may only be applicable to the test since on the frontend it will still go through the process but I thought it was better to keep in line with tests?